clients/ctl/admin-ui/src/features/common/SearchBar.tsx

```diff
@@ -0,0 +1,38 @@
+import {
+  Input,
+  InputGroup,
+  InputLeftElement,
+  InputProps,
+} from "@fidesui/react";
+
+import { SearchLineIcon } from "~/features/common/Icon";
+
+interface Props extends Omit<InputProps, "onChange"> {
+  search?: string;
+  onChange: (value: string) => void;
+}
+const SearchBar = ({ search, onChange, ...props }: Props) => {
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) =>
```

Comment:
I would suggest implementing a debounce such as the following:

```
const debounceHandleSearchChange = useMemo(
    () => debounce(handleSearchChange, 250),
    [handleSearchChange]
  );

  <Input
            autoComplete="off"
            autoFocus
            borderRadius="md"
            name="search"
            onChange={debounceHandleSearchChange}
            placeholder="Search Integrations"
            size="sm"
            type="search"
          />
```

Reply:
So in this case, we aren't calling the API endpoint on search (the search just filters completely in the frontend). I agree a debounce is a good idea if we are calling the API, but I think for this component, if a debounce is needed, the parent can pass a debounced `onChange` instead of embedding it directly into the logic of `SearchBar.tsx`